WordPress/gutenberg
# [Mobile] Editor settings w/ gallery flag

#### Related PRs

`gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3678
`WordPress-Android`: https://github.com/wordpress-mobile/WordPress-Android/pull/14955
`WordPress-FluxC-Android`: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2039

### Description

This PR adds a prop for initialization with raw editor settings as an unparsed string of JSON. It loads and parses the settings when the editor is initialized (through an initial prop), and also attaches a listener to update the settings, if necessary, when the endpoint response is received.

## How has this been tested?

This can be tested via a Jurassic Ninja site by building the plugin locally, and uploading the zip.

`npm run build:plugin-zip`

<details>
<summary>The block-editor-settings filter may not be wired up to the endpoint this way, so it can also be helpful to make these server-side changes for testing purposes</summary>

```
diff --git a/lib/class-wp-rest-block-editor-settings-controller.php b/lib/class-wp-rest-block-editor-settings-controller.php
index 2e10b78566..07dcd605ba 100644
--- a/lib/class-wp-rest-block-editor-settings-controller.php
+++ b/lib/class-wp-rest-block-editor-settings-controller.php
@@ -48,6 +48,7 @@ class WP_REST_Block_Editor_Settings_Controller extends WP_REST_Controller {
 	 * @return WP_Error|bool True if the request has permission, WP_Error object otherwise.
 	 */
 	public function get_items_permissions_check( $request ) {// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+    return true;
 		if ( current_user_can( 'edit_posts' ) ) {
 			return true;
 		}
```
and
```
diff --git a/lib/compat/wordpress-5.8/block-editor.php b/lib/compat/wordpress-5.8/block-editor.php
index 983889dd3f..6800d839bc 100644
--- a/lib/compat/wordpress-5.8/block-editor.php
+++ b/lib/compat/wordpress-5.8/block-editor.php
@@ -225,6 +225,7 @@ function gutenberg_get_default_block_editor_settings() {
 		'imageEditing'                          => true,
 		'imageSizes'                            => $available_image_sizes,
 		'maxUploadFileSize'                     => $max_upload_size,
+    '__experimentalGalleryRefactor'         => true,
 	);
 	// Theme settings.
```

</details>

#### Testing the remote flag

1. Toggle the remote flag (hard-coded if necessary) to true
2. Open the mobile editor with new or existing post
3. Add a gallery block
4. Tap one of the images of the gallery block

Expect the image to be an image block (i.e. the gallery is the refactored version).

1. Toggle the remote flag to false
2. Open the mobile editor with a new or existing post
3. Add a gallery block
4. Tap one of the images of the gallery block

Expect the image to be a part of the gallery block (i.e. not a separate image block, but instead the original gallery block format).


#### Testing the cached flag

1. Toggle the remote flag (hard-coded if necessary) to true
2. Open the mobile editor with new or existing post
3. Leave the editor
4. Turn on airplane mode (w/ no network connection)
5. Toggle the remote flag to false
6. Open the mobile editor with a new or existing post
7. Add a gallery block
8. Tap one of the images of the gallery block

Expect the image to be an image block (i.e. the gallery is the refactored version).

1. Toggle the remote flag to false
2. Open the mobile editor with a new or existing post
3. Leave the editor
4. Turn on airplane mode (w/ no network connection)
5. Toggle the remote flag to true
6. Open the mobile editor with a new or existing post
7. Add a gallery block
8. Tap one of the images of the gallery block

Expect the image to be a part of the gallery block (i.e. not a separate image block, but instead the original gallery block format).


## Screenshots 

## Types of changes





## Checklist:
